### PR TITLE
A page which list all categories

### PR DESCRIPTION
**Pull request by Hakimba**
This PR add a page, accessible for all type of users, which list all the categories. As i don't really know what you expected this is a very minimal stuff, for example the view is basically a part of the `categories` view from `admin_views.ml`. I also create a `catefory_views.ml` file, and i move the Category module in it.

I'm looking forward your thoughts and suggestions !

Screenshots of the page : 

For an admin user

Capture d’écran 2022-06-01 à 11 16 15

For the others type of users : 

Capture d’écran 2022-06-01 à 11 21 17

### CODE REVIEW

**Review by aguillon — CHANGES_REQUESTED** [review]
LGTM except for a very small typo. Sorry.

File: app/services/category_services.mli [hunk 1]
```diff
@@ -0,0 +1,4 @@
+open Lib_service
+
+(** Provide a page which list all available categories for users **)
```
[the remark this answers is not in the document]
Comment (Hakimba) [reply app/services/category_services.mli:3]: i didn't know, thanks

[the remark this answers is not in the document]
Comment (xvw) [reply app/services/category_services.mli]: (In fact, "you have to finish with one star")

**Comment by xvw** [comment]
Ping @gr-im 
Thanks!

File: app/views/category_views.mli [hunk 2]
```diff
@@ -0,0 +1,16 @@
+module Category :
```
Comment (gr-im) [inline app/views/category_views.mli:1]: I suggest to hide this module. 
(TBH, I'm not sure this submodule is useful here).

**Comment by Hakimba** [comment]
Hi, sorry for the huge delay (I'll try to be more diligent in the future).

I removed the submodule and i've hidden the functions which aren't used outside of category_views. I upgraded the view of the category page, now we see the number of topics associated to their categories, if a category doesn't have any topic, i do not display it.

The numbers are clickable, if i click on the number of topics of a category **C**, it redirects to `topic/by/category:C`

I'm looking forward your thoughts and suggestions (even nitpicking ones, i'm a noob).

File: app/views/category_views.ml [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+                      [ txt "Catégories existantes" ]
+                  ; all categories
+                  ]
+              ]
+          ]
+      ]
+;;
+
+let category_topics_count_line cnt_topic_by_category =
+  let (category_name,category_descr,topic_cnt) = cnt_topic_by_category in
```
Comment (gr-im) [inline app/views/category_views.ml:127]: I suggest to change in favor of: `category_topics_count_line (name, desc, counter) = ...`

File: app/views/category_views.mli [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+val all : Models.Category.t list -> [> Html_types.table ] Tyxml_html.elt
+
+(** a view for listing all categories **)
+val categories :
+  ?flash_info:Models.Flash_info.t ->
+  ?user:Models.User.t ->
+  Models.Category.t list -> [> Html_types.html ] Tyxml_html.elt
+
+(** a view for listing all categories grouped by their number of topics **)
+val categories_by_topics_count :
```
Comment (gr-im) [inline app/views/category_views.mli:11]: Names was not clear (and having `categories` as a prefix in a module called `Category` seems a little bit weird).
What is the difference between `all` and `categories`?

**Review by gr-im — CHANGES_REQUESTED** [review]
It is almost perfect, thanks a lot.

File: app/views/category_views.mli (hunk 4 above)
Comment (Hakimba) [reply app/views/category_views.mli:11]: Your right its weird, and i forgot to remove categories, this view is useless now. (it was my previous view).

Comment (Hakimba) [reply app/views/category_views.mli:11]: so, can i rename "category_line" too ? (which was already there before my PR), to "line"

**Comment by gr-im** [comment]
Thanks a lot!

**Pull request merged**